osvimdriver/openstack/heat/template.py

```diff
@@ -23,6 +25,20 @@ def __filter_from_dictionary(self, parameters, properties_dict):
             if k in properties_dict:
                 used_properties[k] = properties_dict[k]
         return used_properties
+    
+    def filter_password_from_dictionary(self, heat_template_str):
+        heat_tpl = yaml.safe_load(heat_template_str)
+        if 'resources' in heat_tpl:
+            resources = heat_tpl['resources'] 
+            user_data = resources['apache_server']['properties']['user_data']
```

Comment:
This code is very specific to apache_server properties. What if there are any other types of servers with password property in it. Please generalise the code to be applicable for all data in the the heat template.

Reply:
Generalised the code to be applicable for all data in the the heat template.